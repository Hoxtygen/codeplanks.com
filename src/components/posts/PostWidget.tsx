import Link from "next/link";
import React, { useState, useEffect } from "react";
import { getRecentPosts, getSimilarPosts } from "../../services";
import { BlogError, PostWidgetProps, PWidget } from "../../typedefs";
import { formatDate } from "../../utils";

export default function PostWidget({ categories, slug }: PostWidgetProps) {
  const [relatedPosts, setRelatedPosts] = useState<PWidget[]>([]);
  const [error, setError] = useState<BlogError | null>(null);
  useEffect(() => {
    if (slug) {
      getSimilarPosts(categories!, slug)
        .then((result) => setRelatedPosts(result))
        .catch((err: BlogError) => setError(err));
    } else {
      getRecentPosts()
        .then((result) => setRelatedPosts(result))
        .catch((err: BlogError) => setError(err));
    }
  }, [slug]);

  return (
    <div className="bg-white shadow-lg rounded-lg p-8 mb-8">
      <h3 className="text-2xl mb-4 font-semibold border-b pb-4">
        {slug ? "Related Posts" : "Recent Posts"}{" "}
      </h3>
      {error && <p>{error.error}</p>}
      {relatedPosts.map((post) => (
        <div key={post.title} className="flex items-center w-full">
          <div className="w-16 flex-none">
            <img
              src={post.featuredImage.url}
              alt=""
              height="60px"
              width="60px"
              className="align-middle rounded-full"
            />
          </div>
          <div className="flex-grow ml-4">
            <p className="text-gray-500 font-xs">
              {formatDate(post.createdAt)}
            </p>
            <Link
              key={post.title}
              href={`/post/${post.slug}`}
              className="text-md"
            >
              {post.title}
            </Link>
          </div>
        </div>
      ))}
    </div>
  );
}
