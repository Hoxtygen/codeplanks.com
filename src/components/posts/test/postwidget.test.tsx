import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

import PostWidget from "../PostWidget";

describe("PostWidget", () => {
  it("should render the PostWidget", () => {
    const { container } = render(<PostWidget />);

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
