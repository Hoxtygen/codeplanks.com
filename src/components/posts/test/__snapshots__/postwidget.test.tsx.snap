// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostWidget should render the PostWidget 1`] = `
<div>
  <div
    class="bg-white shadow-lg rounded-lg p-8 mb-8"
  >
    <h3
      class="text-xl mb-8 font-semibold border-b pb-4"
    >
      Recent Posts
       
    </h3>
  </div>
</div>
`;
